lib/ansible/modules/cloud/vmware/vmware_host_service_manager.py

```diff
@@ -0,0 +1,194 @@
+  state:
+    description:
+    - Desired state of service.
+    - "State value 'start' and 'present' has same effect."
+    - "State value 'stop' and 'absent' has same effect."
+    choices: [ absent, present, restart, start, stop ]
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Start ntpd service setting for all ESXi Host in given Cluster
+  vmware_host_service_manager:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    cluster_name: cluster_name
+    service_name: ntpd
+    state: present
+
+- name: Start ntpd setting for an ESXi Host
+  vmware_host_service_manager:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    service_name: ntpd
+    state: present
+
+- name: Stop ntpd setting for an ESXi Host
+  vmware_host_service_manager:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    service_name: ntpd
+    state: absent
+'''
+
+RETURN = r'''#
+'''
+
+try:
+    from pyVmomi import vim, vmodl
+except ImportError:
+    pass
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.vmware import vmware_argument_spec, PyVmomi
+from ansible.module_utils._text import to_native
+
+
+class VmwareServiceManager(PyVmomi):
+    def __init__(self, module):
+        super(VmwareServiceManager, self).__init__(module)
+        cluster_name = self.params.get('cluster_name', None)
+        esxi_host_name = self.params.get('esxi_hostname', None)
+        self.options = self.params.get('options', dict())
+        self.hosts = []
+        if cluster_name:
+            cluster_obj = self.find_cluster_by_name(cluster_name=cluster_name)
+            if cluster_obj:
+                self.hosts = [host for host in cluster_obj.host]
+            else:
+                module.fail_json(changed=False, msg="Cluster '%s' not found" % cluster_name)
+        elif esxi_host_name:
+            esxi_host_obj = self.find_hostsystem_by_name(host_name=esxi_host_name)
+            if esxi_host_obj:
+                self.hosts = [esxi_host_obj]
+            else:
+                module.fail_json(changed=False, msg="ESXi '%s' not found" % esxi_host_name)
+        self.desired_state = self.params.get('state')
+        self.service_name = self.params.get('service_name')
+        self.results = {}
+
+    def service_ctrl(self):
+        changed = False
+        host_service_state = []
+        for host in self.hosts:
+            actual_service_state = self.check_service_state(host=host, service_name=self.service_name)
+            host_service_system = host.configManager.serviceSystem
+            if host_service_system:
+                service_state = False
```

Comment:
Personally i'd like it better if this was `changed_state`, for instance, as that better indicates the purpose of this variable, looking at the code

Reply:
OK.